packages/medusa/src/services/shipping-option.ts

```diff
@@ -328,10 +338,10 @@ class ShippingOptionService extends TransactionBaseService<ShippingOptionService
 
       const created = await methodRepo.save(method)
 
-      return methodRepo.findOne({
+      return (await methodRepo.findOne({
         where: { id: created.id },
         relations: ["shipping_option"],
-      }) as unknown as ShippingMethod
+      })) as ShippingMethod
```

Comment:
question(non-blocking): is `as ShippingMethod` nessecary

Reply:
Because the `findOne` method return either the entity or undefined, but in that context where we have just created the entity, we know that it can't return undefined. wdyt?